### PR TITLE
feat(EU): Support setting the charging current

### PR DESCRIPTION
**Pull request by jrnk**
Following up with this to support https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/commit/632951ff7a970541dec7bcde0960c47bcbaa71a2 

Should solve https://github.com/Hyundai-Kia-Connect/kia_uvo/issues/864

Note: I've translated all but the Chinese (that was missing more translation keys), but the indentation seems to be different for every localization which was a bit cumbersome, but I didn't want to clutter this commit with formatting.

### CODE REVIEW

**Comment by cdnninja** [comment]
Looks good.  Would it be worth adding a sensor to display it? Either a number slider so you can edit or just a sensor to display.  

Number with edit may be hard since only a few values work, I guess drop down menu could be an option too.

**Comment by jrnk** [comment]
Yes, good idea, will add it!

**Comment by cdnninja** [comment]
Same pr or different?

**Comment by cdnninja** [comment]
Also the manifest file needs to be updated to the new api version.

**Comment by jrnk** [comment]
> Also the manifest file needs to be updated to the new api version.

All will be in this PR, already bumped the version in the manifest

**Comment by jrnk** [comment]
@cdnninja added the sensor, also needed to add a mapping function for it, if you have another suggestion to do this: please shoot, this is new for me ;)

**Comment by jrnk** [comment]
> Number with edit may be hard since only a few values work, I guess drop down menu could be an option too.

The selector was already added in `services.yml`, does it need more for it to work?

**Comment by cdnninja** [comment]
> > Number with edit may be hard since only a few values work, I guess drop down menu could be an option too.
> 
> 
> 
> The selector was already added in `services.yaml`, does it need more for it to work?
> 
> 
> 
> 


Services work but people often want it on the dashboard.  We can do without though, you have done the heavy lift.

**Comment by cdnninja** [comment]
> > Number with edit may be hard since only a few values work, I guess drop down menu could be an option too.
> 
> 
> 
> The selector was already added in `services.yaml`, does it need more for it to work?
> 
> 
> 
> 

Mapping is often done as enum in the constants file in both the api and home assistant files.

File: custom_components/kia_uvo/sensor.py [hunk 1]
```diff
@@ -240,6 +240,13 @@
         native_unit_of_measurement=PERCENTAGE,
         device_class=SensorDeviceClass.BATTERY,
     ),
+    SensorEntityDescription(
+        key="ev_charging_current",
+        name="Charging current limit",
```
Comment (cdnninja) [inline custom_components/kia_uvo/sensor.py:245]: We use capitals for each letter for exposed names.

File: custom_components/kia_uvo/sensor.py [hunk 2]
```diff
@@ -372,3 +379,8 @@ def unique_id(self):
     @property
     def unit_of_measurement(self):
         return UnitOfTime.DAYS
+
+
+def charging_current_mapper(value):
+    mapping = {1: 100, 2: 90, 3: 60}
```
Comment (cdnninja) [inline custom_components/kia_uvo/sensor.py:385]: This should move to constants.

File: custom_components/kia_uvo/services.yaml [hunk 3]
```diff
@@ -226,3 +226,21 @@ set_charge_limits:
...
+      selector:
+        device:
+          integration: kia_uvo
+    level:
+      required: true
+      selector:
+        select:
+          options:
+            - label: "100%"
+              value: 1
```
Comment (cdnninja) [inline custom_components/kia_uvo/services.yaml:242]: Not sure why but the CI isn't happy with this: https://github.com/Hyundai-Kia-Connect/kia_uvo/actions/runs/9652140110/job/26621566236

**Review by cdnninja — CHANGES_REQUESTED** [review]
Here is a bit of feedback. It is very close! You have learned quick.   Help on this integration is appreciated this I no longer have one of these cars.

**Comment by jrnk** [comment]
@tobixx yes saw that! Fixed in https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/pull/581

**Comment by jrnk** [comment]
@tobixx ah I missed you could not set it in the app, then it's not available for the EV6 I guess. For an EV9 you can control this current in the app on the "EV service" page. 

image

**Pull request merged**